### PR TITLE
Feat/platypus [TKR-274]

### PR DESCRIPTION
**Pull request by dextracker**
## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinPlatypus.sol [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        internal
+        returns (uint256 boughtAmount)
+    {
+        // Decode the bridge data to get the Platypus metadata.
+        PlatypusBridgeData memory data = abi.decode(bridgeData, (PlatypusBridgeData));
+        sellToken.approveIfBelow(data.platypusAddress, sellAmount);
+
+        uint256 beforeBalance = buyToken.balanceOf(address(this));
+        uint256 minAmountOut = _swapQuote(data);
+        (bool success, bytes memory resultData) =
```
Comment (mzhu25) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinPlatypus.sol:77]: I believe the swap function returns the output amount

File: packages/asset-swapper/src/utils/market_operation_utils/types.ts [hunk 2]
```diff
@@ -134,6 +135,9 @@ export enum CurveFunctionSelectors {
     // Nerve BSC, Saddle Mainnet, Synapse
     swap = '0x91695586', // swap(uint8,uint8,uint256,uint256,uint256)
     calculateSwap = '0xa95b089f', // calculateSwap(uint8,uint8,uint256)
+    // Platypus
```
Comment (mzhu25) [inline packages/asset-swapper/src/utils/market_operation_utils/types.ts:138]: I thought platypus wasn't a curve forK?

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/platypus sighash [hunk 3]
```diff
@@ -0,0 +1,45 @@
+Sighash   |   Function Signature
```
Comment (mzhu25) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/platypus sighash:1]: delete this file?

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinPlatypus.sol (hunk 1 above)
Comment (dextracker) [reply contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinPlatypus.sol:77]: forgot it auto requests you to review this when i make a pr, im moving the new mixin into asset swapper and the platypus as a curve source was just testing from a bit ago to see if i could squeeze it into our already existing interfaces.

Comment (dextracker) [reply contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinPlatypus.sol:77]: and yeah, i can pass in a min amount of 0 here that will be removed

**Pull request closed without merging**